lib/Interpreter/CppInterOp.cpp

```diff
@@ -3307,22 +3307,42 @@ namespace Cpp {
     return PI->getNameAsString();
   }
 
-  void GetBinaryOperator(TCppScope_t scope, enum BinaryOperator op,
-                         std::vector<TCppFunction_t>& operators) {
-    Decl* D = static_cast<Decl*>(scope);
-    auto* DC = llvm::dyn_cast<DeclContext>(D);
-    Scope* S = getSema().getScopeForContext(DC);
-    if (!S)
-      return;
-
-    clang::UnresolvedSet<8> lookup;
+  OperatorArity getOperatorArity(TCppFunction_t op) {
+    Decl* D = static_cast<Decl*>(op);
+    if (auto* FD = llvm::dyn_cast<FunctionDecl>(D)) {
+      if (FD->isOverloadedOperator()) {
+        switch (FD->getOverloadedOperator()) {
+#define OVERLOADED_OPERATOR(Name, Spelling, Token, Unary, Binary,            \
+                              MemberOnly)                                      \
+    case OO_##Name:                                                            \
+      if ((Unary) && (Binary))                                                 \
+        return kBoth;                                                          \
+      if (Unary)                                                               \
+        return kUnary;                                                         \
+      if (Binary)                                                              \
+        return kBinary;
+#include "clang/Basic/OperatorKinds.def"
+        default:
+          break;
+        }
+      }
+    }
+    return kNone;
```

Comment:
That should be unreachable, right? What does it mean `arity::kNone`? An operator without arguments? I do not think that's possible.

Reply:
Yes, should be unreachable.
It is for cases when the argument is something other than an operator or a function that is not an operator.

Reply:
Shouldn't that be an assert or an early exit?

Reply:
I will say this is simpler. Assert will crash the program. In this case, the user can handle `kNone` as an error.

Also if I remove the return, clang-tidy will get angry.

Let me know if you still think we should remove `kNone`. I can do that, but not sure how I would satisfy clang-tidy's message of `Non-void function does not return a value in all control paths`.

Reply:
You should remove it. You can return `~0U`

Reply:
Compiler error:
```bash
/home/vipul-cariappa/Workspace/cpp-py/compiler-research/CppInterOp/lib/Interpreter/CppInterOp.cpp:3339:12: error: invalid conversion from ‘unsigned int’ to ‘Cpp::OperatorArity’ [-fpermissive]
 3339 |     return ~0U;
      |            ^~~
      |            |
      |            unsigned int
```

Should I rename it to `kInvalid` instead?